### PR TITLE
Resurrect the option to force push the Docker image

### PR DESCRIPTION
**Pull request by huydhn**
IIRC, we used to have this option, but nothing was using it.  I have found an use case for this in https://github.com/pytorch/executorch/pull/1335#discussion_r1448247422 where ExecuTorch team copies the content of `examples/arm` into `.ci/docker` before starting the build https://github.com/pytorch/executorch/blob/main/.ci/docker/build.sh#L40-L46:

* The docker image was rebuilt, but its hash tag remained the same because the copy step above is not version tracked.
* The new image was not pushed.

I have tried to move `examples/arm` into `.ci/docker` and setup softlinks, but the result looks weird https://github.com/pytorch/executorch/pull/1574#discussion_r1447762323, so I'll probably drop that approach.

The remaining option here is to add a `force-push` flag that can be used to, well, force push the image even if its hash tag already exists.

### CODE REVIEW

File: .github/actions/calculate-docker-image/action.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ inputs:
   push:
     description: If set to true, push the image to ECR.
     required: false
+  force-push:
+    description: If set to true, always push the image to ECR even if it exists.
+    required: false
```
Comment (atalman) [inline .github/actions/calculate-docker-image/action.yml:31]: Should we add here ?
```
default: false
```

**Review by atalman — APPROVED** [review]
LGTM. Just add default value

**Comment by guangy10** [comment]
> The remaining option here is to add a force-push flag that can be used to, well, force push the image even if its hash tag already exists.

Catch up late, sorry. Does it mean the docker image will be rebuilt everytime when arm is running their job? The reason we decided to move the expensive package downloading and installation to docker is to avoid that kind of overhead. Just want to confirm we are not adding it back.

**Comment by huydhn** [comment]
> Catch up late, sorry. Does it mean the docker image will be rebuilt everytime when arm is running their job? The reason we decided to move the expensive package downloading and installation to docker is to avoid that kind of overhead. Just want to confirm we are not adding it back.

Rebuilding the Docker image only happen when ARM's setup scripts are updated according to the logic in https://github.com/pytorch/executorch/blob/main/.github/workflows/docker-builds.yml#L10-L11, so there would not be any overhead.  The flag here only means that when the image is rebuilt, it will always be updated onto AWS ECR (if the build finishes successfully).

On the other hand, we need to keep in mind the different behavior of the docker build workflow.  For example, the Docker images are always rebuilt daily.  Previously, the newly built images weren't pushed upstream if their tags were unchanged.  Now, the push will always happen and the image will be refreshed daily.

**Pull request merged**